src/js/form.js

```diff
@@ -0,0 +1,122 @@
+                input: $(this.form[0][2]),
+                border: $($(".border--input")[2]),
+                errorMessage: "Informe um número válido.",
+            }
+        };
+    }
+
+    submit() {
+        if (this.validate()) {
+            const body = {
+                "name": this.inputs['name'].input.val(),
+                "email": this.inputs['email'].input.val(),
+                "phone": this.inputs['cellphone'].input.val(),
+            }
+
+            $.ajax({
+                type: "POST",
+                url: process.env.LEAD_URL,
+                data: JSON.stringify(body),
+                success: this.onSuccess.bind(this),
+                error: this.onError.bind(this)
+            });
+        };
+    }
+
+    validate() {
+        var valid = true;
+        const inputNames = Object.keys(this.inputs);
+
+        inputNames.forEach((inputName) => {
+            if ((inputName != 'cellphone' && this.isEmpty(this.inputs[inputName].input.val())) ||
+                (inputName == 'cellphone' && this.inputs['cellphone'].input.val() && phoneParse(this.inputs['cellphone'].input.val()) == null)) {
+
+                this.inputError(this.inputs[inputName]);
+                this.errorMessage(this.inputs[inputName].errorMessage);
+                this.inputs[inputName].error = true;
+                valid = false;
+            }
+        });
+
+        return valid;
+    }
+
+    isEmpty(string) {
+        const regex = new RegExp(/^\s+$/);
+        return string == "" || regex.test(string);
+    }
+
+    inputError(element) {
+        element.border.css("border", "1px solid #E04646");
+        element.input.css("border", "1px solid #E04646");
+    }
+
+    errorMessage(msg) {
+        let message = $(".register__message");
+        message.css("display", "block");
+        message.html(msg);
+    }
+
+    reset(element) {
+        this.inputs[element.name].border.css("border", "1px solid white");
+        $(element).css("border", "1px solid white");
+        $(".register__message").css("display", "none");
+        this.inputs[element.name].error = false;
+    }
+
+    onError(_) {
+        this.errorMessage("Ocorreu um erro inesperado, tente novamente mais tarde.");
+    }
+
+    onSuccess(response) {
+        if (response.status == "success") {
+            const title = $("#register .title");
+            const subtitle = $(".register__request-account");
+
+            $(title[1]).html("<span class='marked'> Cadastrado com sucesso! </span>");
+            this.form.css("display", "none");
+
+            if (subtitle.css("display") == "none") {
+                subtitle.css("display", "block");
+            }
+
+            subtitle.html("Ficamos felizes com o seu interesse! Toda nova atualização, lembraremos de enviar à você!");
+
+        } else {
+            this.errorMessage("Ocorreu um erro inesperado, tente novamente mais tarde.");
+        }
+    }
+}
+
+const formControl = new FormControl("form");
```

Comment:
Se fizer as modificações que eu sugeri, nem precisamos mais declarar uma constante para armazenar o objeto do FormControl, não poluindo o namespace, podemos só instanciar que ele vai gerenciar bonitinho...

```suggestion
new FormControl("form");
```